### PR TITLE
Better severity calculation when detekt config provided

### PR DESCRIPTION
**Pull request by os97673**
a fix for  #504 

if user provided a config then let's use levels from this config instead of all-warnings or all-errors.

### CODE REVIEW

File: src/main/kotlin/io/gitlab/arturbosch/detekt/idea/DetektAnnotator.kt [hunk 1]
```diff
@@ -75,5 +77,21 @@ class DetektAnnotator : ExternalAnnotator<PsiFile, List<Finding>>() {
...
     }
 
+    private fun getSeverity(finding: Finding, hasCustomConfig: Boolean, treatAsError: Boolean): HighlightSeverity {
+        if (treatAsError) {
+            return HighlightSeverity.ERROR
+        }
+
+        if (!hasCustomConfig) {
+            return HighlightSeverity.WARNING
+        }
```
Comment (schalkms) [inline src/main/kotlin/io/gitlab/arturbosch/detekt/idea/DetektAnnotator.kt:87]: Don't quite understand this clause? May I ask you to please elaborate what's the reason for it.

Comment (os97673) [reply src/main/kotlin/io/gitlab/arturbosch/detekt/idea/DetektAnnotator.kt:87]: I wanted to limit changes to the case when user does provide a custom detekt config.  WDYT?

Comment (schalkms) [reply src/main/kotlin/io/gitlab/arturbosch/detekt/idea/DetektAnnotator.kt:87]: Yeah, I can see that. But with no provided custom config every issue is reported as **warning**. I don’t get why users would want that. We didn’t have this in previous versions.

Comment (os97673) [reply src/main/kotlin/io/gitlab/arturbosch/detekt/idea/DetektAnnotator.kt:87]: This is how the plugin behaves now so I'm just preserving current behavior.

**Review by schalkms — APPROVED** [review]

**Pull request merged**